### PR TITLE
Enable VideoPress block on iOS

### PR DESCRIPTION
**Pull request by SiobhyB**
With this PR, the VideoPress block has been registered behind a DEV flag. The block itself is not complete or fully functional, but is intended to serve as a useful placeholder to be iterated on.

## Related PRs

* `Gutenberg`: https://github.com/WordPress/gutenberg/pull/47836
* `Gutenberg Mobile`: https://github.com/wordpress-mobile/gutenberg-mobile/pull/5459
* `Jetpack`: https://github.com/Automattic/jetpack/pull/28812
* `Android`: https://github.com/wordpress-mobile/WordPress-Android/pull/17928

## Testing

Please refer to [the Gutenberg Mobile PR](https://github.com/wordpress-mobile/gutenberg-mobile/pull/5459) as the "central PR" with the most up-to-date testing instructions.

<hr />

## Regression Notes

1. Potential unintended areas of impact
2. What I did to test those areas of impact (or what existing automated tests I relied on)

As this is a new block that's been registered behind a DEV flag, it's hoped it won't have any impact on other blocks or areas of the editor. That said, it's always best to be careful and I manually smoke tested some other blocks (including the core video block) to check for any obvious issues.

3. What automated tests I added (or what prevented me from doing so)

No automated tests for now as this is the first step in a larger project. Tests related to this new block will come as it's worked on and functionality is added.

<hr />

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/Models/Blog.m [hunk 1]
```diff
@@ -713,6 +715,11 @@ - (BOOL)supportsTiledGallery
     return self.isHostedAtWPcom;
 }
 
+- (BOOL)supportsVideoPress
+{
+    return self.isHostedAtWPcom;
```
Comment (SiobhyB) [inline WordPress/Classes/Models/Blog.m:720]: Enabled only on simple WordPress.com sites for now and added [a task to the project board](https://github.com/orgs/wordpress-mobile/projects/166/views/1?layout=board) for this to eventually be changed to match the web.

File: Podfile [hunk 2]
```diff
@@ -91,7 +91,7 @@ def shared_style_pods
 end
 
 def gutenberg_pods
-  gutenberg tag: 'v1.88.0'
+  gutenberg commit: '0b4b05d499197825893260921c662ed3d45a6f33'
```
Comment (fluiddot) [inline Podfile]: We'd need to update this reference to the latest commit: `364d743d55c79c0eb462a1df79520173bcb8aab9`.

**Review by fluiddot — CHANGES_REQUESTED** [review]
I tested the installable build and noticed that I can add the VideoPress block to a post/page. However, if this block is only meant to be available in development mode (i.e. using `__DEV__` flag), it should not be available in the installable builds.

**Comment by SiobhyB** [comment]
> I tested the installable build and noticed that I can add the VideoPress block to a post/page. However, if this block is only meant to be available in development mode (i.e. using __DEV__ flag), it should not be available in the installable builds.

Thanks for spotting this! I'll look into what's causing this. 🧐

**Comment by SiobhyB** [comment]
@fluiddot, I've added a fix for this with [this commit](https://github.com/wordpress-mobile/gutenberg-mobile/pull/5459/commits/2209d48ca436bcd8cf4f610c08a4fa4e13960e6a), thanks again for spotting this! No rush to re-review, I don't plan to begin the merge domino until next week 🙇‍♀️

**Review by fluiddot — CHANGES_REQUESTED** [review]
> I tested the installable build and noticed that I can add the VideoPress block to a post/page. However, if this block is only meant to be available in development mode (i.e. using __DEV__ flag), it should not be available in the installable builds.

As I mentioned in https://github.com/wordpress-mobile/gutenberg-mobile/pull/5459#pullrequestreview-1295417190, the VideoPress block is no longer available for insertion, however, it's still being rendered as a regular block instead of an unsupported block.

**Comment by SiobhyB** [comment]
Noting that I'm working to fix the failing builds.

**Comment by SiobhyB** [comment]
@fluiddot, the newest installable builds have finished generating now, whenever you have availability for taking another look. Thanks in advance! :)

**Review by fluiddot — APPROVED** [review]
LGTM 🎊 !

I checked that the VideoPress block is displayed as unsupported and that can be edited via UBE (Unsupported Block Editor) on Atomic sites.

**NOTE:** On Simple sites when using UBE, it shows a warning due to the block not being available on the site. This is expected as the block on the web hasn't been released yet to users.

_Tested build `pr20100-17ddbb8` on an iPhone 11 (iOS 15.4)._

**Pull request merged**